src/deprecates.jl

```diff
@@ -48,3 +48,15 @@ findat(a::AbstractArray, b::AbstractArray) = findat!(Array{Int}(size(b)), a, b)
 @deprecate df_residual(obj::StatisticalModel) dof_residual(obj)
 
 @deprecate_binding WeightVec Weights
+
+using Base.Random: RangeGenerator
+@deprecate_binding RandIntSampler RangeGenerator
+@deprecate RangeGenerator(K::Int) RangeGenerator(1:K)
```

Comment:
If I don't use `@deprecate_binding`, the depwarn is not triggered, but I need to use `const RandIntSampler = RangeGenerator`, and then the methods are added to `RangeGenerator`. So that's equivalent to the current situation of the PR. If I only define `RandIntSampler` as a function, the places where it's used as a type will break (there's one case in Distributions, see https://github.com/JuliaStats/Distributions.jl/issues/630). We can certainly do that, the question is whether it's better or worse than the type piracy.